internal/core/tx.py

```diff
@@ -1,38 +1,72 @@
+        from_address (_str_): Name or address of private key with which to sign.
+        sequence (_uint_): The sequence number of the signing account (offline mode only)
+        gas (str, optional): gas limit to set per-transaction;
+        set to "auto" to calculate sufficient gas automatically (default 200000).
+
+    Returns:
+        _tuple_: (boolean, json|str)
+    """
     try:
-        command = f"{DAEMON} tx sign {HOME}/{unsigned_file_name} --from {from_address} --chain-id {CHAINID} --keyring-backend test --home {DAEMON_HOME}-1 --node {RPC} --signature-only=false --sequence {sequence} --gas {gas} --output json"
-        signTx, signTxerr = exec_command(command)
-        if len(signTxerr):
-            return False, signTxerr
-        return True, json.loads(signTx)
-    except Exception as e:
-        return False, e
+        command = f"""{DAEMON} tx sign {HOME}/{unsigned_file_name} --from {from_address} \
+--chain-id {CHAINID} --keyring-backend test \
+--home {DAEMON_HOME}-1 --node {RPC} --signature-only=false \
+--sequence {sequence} --gas {gas} --output json"""
+        sign_tx, sign_tx_err = exec_command(command)
+        if len(sign_tx_err) != 0:
+            return False, sign_tx_err
+        return True, json.loads(sign_tx)
+    except Exception as error:  # pylint: disable=broad-except
+        return False, error
 
 
-# The function `tx_broadcast` will call cosmos-sdk command `tx broadcast` and the returns the response in json format.
 def tx_broadcast(signed_file, gas, broadcast_mode="sync"):
+    """
+    Broadcast transactions created with the --generate-only
+    flag and signed with the sign command. Read a transaction from [file_path] and
+    broadcast it to a node.
+
+    Args:
+        signed_file (_str_): file path to store signed transactions.
+        gas (_int_): gas limit to set per-transaction; set to "auto"
+        to calculate sufficient gas automatically (default 200000)
+        broadcast_mode (str, optional): Transaction broadcasting mode (sync|async|block)
+        (default "sync").
+
+    Returns:
+        _tuple_: (boolean, json|str)
+    """
     try:
         if broadcast_mode == "block":
             logging.info("Waiting for transaction for being broadcasted")
-        command = f"{DAEMON} tx broadcast {HOME}/{signed_file} --output json --chain-id {CHAINID} --gas {gas} --node {RPC} --broadcast-mode {broadcast_mode}"
-        broadcastTx, broadcasterr = exec_command(command)
-        broadcastTx = json.loads(broadcastTx)
-        if len(broadcasterr):
-            return False, broadcasterr
-        elif broadcastTx["code"] != 0:
-            return False, broadcastTx
-        return True, broadcastTx
-    except Exception as e:
-        return False, e
+        command = f"""{DAEMON} tx broadcast {HOME}/{signed_file} --output json \
+--chain-id {CHAINID} --gas {gas} --node {RPC} --broadcast-mode {broadcast_mode}"""
+        broadcast_tx, broadcast_err = exec_command(command)
```

Comment:
may be let's convert `exec_command` to return error or `json.loads()` directly. We are repeating this everywhere. We can remove `process_response` too

Reply:
The new commit has changes reflecting the aforementioned comment.